src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2194 @@
+            var model = comp.GetSemanticModel(tree);
+
+            var nodes = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>();
+            Assert.Equal(2, nodes.Count());
+
+            foreach (var assign in nodes)
+            {
+                var node = assign.Left;
+                Assert.Equal("P1", node.ToString());
+                Assert.Equal("System.Int32 C.P1 { get; set; }", model.GetSymbolInfo(node).Symbol.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void ModReqOnAccessorParameter()
+        {
+            string il = @"
+.class public auto ansi beforefieldinit C extends System.Object
+{
+    .method public hidebysig specialname newslot virtual instance void set_Property ( int32 modreq(System.Runtime.CompilerServices.IsInitOnly) 'value' ) cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .property instance int32 Property()
+    {
+        .set instance void C::set_Property(int32 modreq(System.Runtime.CompilerServices.IsInitOnly))
+    }
+}
+
+.class public auto ansi sealed beforefieldinit System.Runtime.CompilerServices.IsInitOnly extends System.Object
+{
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+}
+";
+
+            string source = @"
+public class Derived : C
+{
+    public override int Property { set { throw null; } }
+}
+public class Derived2 : C
+{
+    public override int Property { init { throw null; } }
+}
+";
+
+            var reference = CreateMetadataReferenceFromIlSource(il);
+            var comp = CreateCompilation(source, references: new[] { reference }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,25): error CS0569: 'Derived.Property': cannot override 'C.Property' because it is not supported by the language
+                //     public override int Property { set { throw null; } }
+                Diagnostic(ErrorCode.ERR_CantOverrideBogusMethod, "Property").WithArguments("Derived.Property", "C.Property").WithLocation(4, 25),
+                // (8,25): error CS0569: 'Derived2.Property': cannot override 'C.Property' because it is not supported by the language
+                //     public override int Property { init { throw null; } }
+                Diagnostic(ErrorCode.ERR_CantOverrideBogusMethod, "Property").WithArguments("Derived2.Property", "C.Property").WithLocation(8, 25)
+                );
+        }
+
+        [Fact]
+        public void TestSyntaxFacts()
+        {
+            Assert.True(SyntaxFacts.IsAccessorDeclaration(SyntaxKind.InitAccessorDeclaration));
+            Assert.True(SyntaxFacts.IsAccessorDeclarationKeyword(SyntaxKind.InitKeyword));
+        }
```

Comment:
Could you add a test for the IL of an emitted init-only autoprop? I wanted to confirm that the backing field was still emitted as `readonly` even though the property has a setter.

Reply:
That is verified by `VerifyPESymbols_Property` which looks at source and PE symbol for the backing field.

I don't think we can verify that the backing field is emitted as `readonly` with `VerifyIL`. Did you mean some other verification method?

---
In reply to: [413382444](https://github.com/dotnet/roslyn/pull/43275#discussion_r413382444) [](ancestors = 413382444)

Reply:
The `VerifyTypeIL` method should allow you to dump the IL for the whole type, which should show the `initonly` flag in metadata.

Reply:
> The VerifyTypeIL method should allow you to dump the IL for the whole type, which should show the initonly flag in metadata. 

I would prefer us to use symbols to verify what we emitted rather than dumping type IL just to verify presence/lack of a modreq. 

---
In reply to: [413476781](https://github.com/dotnet/roslyn/pull/43275#discussion_r413476781) [](ancestors = 413476781)

Reply:
Not the modreq, I meant the actual `initonly` flag on the backing field.

Previously we used the absence of a setter to mean "emit the backing field as `initonly`. I want to make sure the `init` accessor hasn't broken that logic.

Reply:
That said, verifying the PE symbol is probably fine too

Reply:
Cool. I'll stick with the verification on PE symbol then. Thanks

---
In reply to: [413999594](https://github.com/dotnet/roslyn/pull/43275#discussion_r413999594) [](ancestors = 413999594)